### ISSUE TITLE
Assignment within conditional rule?

### ISSUE THREAD

**Issue by bkdotcom** [phpmd/phpmd#393]
I was just on http://www.reddit.com/r/phphelp where a rookie didn't know why his code wasn't working.      They had committed the 'ol `=` vs `==` mistake

``` php
if ($foo = 'bar') {
     echo "Frowny face";
}
```

We've all done it.  It's always more elusive to track down than it should be.

Candidate for a ["design" rule](https://phpmd.org/rules/design.html)?

There are multiple cases where no warning is warranted - such as when getting the result from a function or method:

``` php
while ($row = $result->fetch_row()) {
    // no need to throw a warning about the assignment operator here
}
```

thoughts?

**Comment by flip111** [comment]
Good candidate for a warning with low priority. Would not warn in case where a boolean operator is used such as `===` or `and`.
